cloud/sts.go

```diff
@@ -61,6 +64,15 @@ type AssumeRoleCredentials struct {
 	Expiration      time.Time
 }
 
+// assumeRoleCache holds AWSCredentials for assumed roles.
+var assumeRoleCache = cache.WithOtel(cache.NewTTLInMemory[AssumeRoleCredentials](), "aws-assume-role")
+
+// minAssumeRoleCacheLifetime is the minimum lifetime of an assumed role
+// when retrieved from the cache. This is set low because the cache is currently
+// used in s3 credentials, which only need valid credentials for the
+// beginning of the operation.
```

Comment:
nit: rather than state the current use case as the reason for the constant, it might be preferable to say "This can be used in situations where it's known that the credentials don't need to be used for a long time, such as S3 operations that only need valid credentials for the beginning of the operations". Stating the current implementation isn't usually as helpful as informing future readers on how to use this constant/an example of how it's used.

Reply:
Swapped it to that wording!